python_modules/dagster/dagster/_serdes/serdes.py

```diff
@@ -585,14 +628,56 @@ def after_pack(self, **packed_dict: JsonSerializableValue) -> Dict[str, JsonSeri
         return packed_dict
 
     @property
-    @cached_method
+    @abstractmethod
     def constructor_param_names(self) -> Sequence[str]:
-        return list(signature(self.klass.__new__).parameters.keys())
+        ...
 
     def get_storage_name(self) -> str:
         return self.storage_name or self.klass.__name__
 
 
+T_NamedTuple = TypeVar("T_NamedTuple", bound=NamedTuple, default=NamedTuple)
+
+
+class NamedTupleSerializer(EntitySerializer[T_NamedTuple], Generic[T_NamedTuple]):
```

Comment:
[2] I don't think you need the second `Generic` here?

Reply:
I did this believing that if you wanted to specify a generic on the subclass of this that it needs `Generic` again - will double check